### ISSUE TITLE
Chrome 44.0 issue

### ISSUE THREAD

**Issue by gwixted** [alvarotrigo/fullPage.js#1389]
Have been building a site using fullPage and everything was working great until today (Chrome 44.0 was released yesterday). Now, with autoScrolling enabled (the default behaviour) it appears to jump to the first section each time i scroll (using the mouse) and scrolls through all sections until it reaches the 'next' section. It's even happening on the fullPage demo page.

**Comment by alvarotrigo** [comment]
Can't reproduce it in [the demo page](http://alvarotrigo.com/fullPage/). 
Please provide a reproduction.  Closed until then.

**Comment by alvarotrigo** [comment]
Thanks for reporting it. I'll investigate it now.

**Comment by alvarotrigo** [comment]
Have you tried it on Mac or Windows?
I saw the problem taking place in Mac. Not sure about windows yet.

**Comment by gwixted** [comment]
I'm only seeing it on Mac. Just tested it on Windows (Windows 7, Chrome 44.0) and it appears to be working just fine.

**Comment by alvarotrigo** [comment]
Mmmm weird... are you using mouse or trackpad ?

**Comment by gwixted** [comment]
Mouse on Mac, trackpad on Windows.

**Comment by alvarotrigo** [comment]
Ok thanks. I keep investigating it.

**Comment by alvarotrigo** [comment]
Seems to be a tough one... It seems to me that there's some kind of bug in Chrome with translat3d, but I can not isolate it and reproduce it.

**Comment by alvarotrigo** [comment]
The problem doesn't take place when using keyboard to navigate up or down. 
I've reduced the plugin to the basics (400 lines) and still can not detect where the problem is...

**Comment by alvarotrigo** [comment]
Here I have fullpage.js reduced to 100 lines and being able to reproduce the bug by just scrolling down.
http://jsfiddle.net/6fdLk7fo/

As you can see, the first section will become visible again when scrolling from the 2nd to the 3rd section. 
This shouldn't take place.

**Comment by alvarotrigo** [comment]
A much cleaner reproduction of the issue:
http://jsfiddle.net/9ksx000q/3/

**Comment by alvarotrigo** [comment]
It happens in Windows as well :(

**Comment by mcadam** [comment]
+1 same problem using Mac and trackpad

**Comment by alvarotrigo** [comment]
I opened a question in stackoverflow hoping somebody else can realise about it or propose any solution:
http://stackoverflow.com/questions/31575177/bug-with-translate3d-in-chrome-v-44

**Comment by TravnikovRN** [comment]
+1 victim, Chrome 44 on Win 7

**Comment by alvarotrigo** [comment]
You can temporally solve the problem in your sites by using `css3:false`.

**Comment by gwixted** [comment]
Setting css3:false seems to be a good stopgap solution.

**Comment by alvarotrigo** [comment]
@gwixted but far from ideal.

**Comment by larrybotha** [comment]
setting `css3: false` has a side effect where the last slide doesn't show for me O.o

I'm gonna take a look and see if there's anything I can find to expedite this.

**Comment by larrybotha** [comment]
bug opened before Chrome 44 release... bug made it into production: https://code.google.com/p/chromium/issues/detail?id=506893

**Comment by hexplus** [comment]
Same here: Mac OS X Yosemite 10.10.2 - Chrome 44.0.2403.89 (64-bit)

**Comment by llahnoraa** [comment]
Same here: Mac OS X Yosemite `10.10.4 (14E46)` - Chrome Version `44.0.2403.89 (64-bit)` using Apple TrackPad (not mouse) for iMac.

**Comment by llahnoraa** [comment]
Also, the up/down arrow does work perfectly, FYI.

**Comment by llahnoraa** [comment]
For some reason, Google Canary Version `46.0.2464.0 canary (64-bit)` have the same issue. I wondered if it's TrackPad bug-related?

**Comment by maverickp** [comment]
Same problem with Chrome 44.0 in Linux (with trackpad)

**Comment by llahnoraa** [comment]
The new Chrome update v44.0.2403.107 (64-bit) still doesn't help. Blame on Google...

**Comment by panteng** [comment]
Same problem. Even the demo site of Fullpage.js itself has this problem.

**Comment by alvarotrigo** [comment]
An issue was opened days ago in Google Chromium issues forum: https://code.google.com/p/chromium/issues/detail?id=513206

**Comment by el-rotny** [comment]
+1 Will keep in this update in mind Alvaro. Things might be shifting to https://codereview.chromium.org/1246363002/

**Comment by alvarotrigo** [comment]
Good news folks!
The Chrome bug has been solved in fullpage.js 2.6.9 (https://github.com/alvarotrigo/fullPage.js/commit/b046fafab5fee5aaa3e1343d8bcc8d9b8f817be0).

The bug still there in Chrome and I hope they fix it soon, but now it won't affect us!

**Comment by richgcook** [comment]
Well done, nice one! What was the issue? I think flickity (https://github.com/metafizzy/flickity) has the same problem.

**Comment by alvarotrigo** [comment]
I do not know the cause of the issue. But it seems to work when using `requestAnimFrame` function as you can see in [the code](https://github.com/alvarotrigo/fullPage.js/commit/b046fafab5fee5aaa3e1343d8bcc8d9b8f817be0).

**Comment by richgcook** [comment]
Thanks @alvarotrigo for the help.

**Comment by larrybotha** [comment]
awesome work

a note for coffee users - make sure your explicitly return a value from your onLeave callback, or at least have a return statement. You might be getting false negatives, preventing scrolling with the RAF update.

**Comment by alvarotrigo** [comment]
> a note for coffee users - make sure you explicitly return a value from your onLeave callback, or at least have a return statement. You might be getting false negatives, preventing scrolling with the RAF update.

Could you elaborate that?

**Comment by larrybotha** [comment]
haha, ye, that was probably a little vague.

Coffee by default returns the last statement in methods / functions. If you don't want to return anything, then you need to use `return` on your last line (somewhat unfortunate).

e.g.
```coffee
  onLeaveHandler = () ->
    doSomething()
```

compiles to

```javascript
  onLeaveHandler = function() {
    return doSomething();
  }
```

while 

```coffee
  onLeaveHandler = () ->
    doSomething()
    return
```

compiles to

```javascript
  onLeaveHandler = function() {
    doSomething();
  }
```

In my case, my `doSomething()` was returning `false`, resulting in my slides not scrolling, but only after updating from 2.6.5. I did jump from 2.6.5 to 2.6.9, so the affecting change is possibly somewhere between those versions, and likely not in 2.6.9 specifically.

**Comment by alvarotrigo** [comment]
@larrybotha thanks for the explanation. It is due to [this new feature introduced time ago](https://github.com/alvarotrigo/fullPage.js#cancelling-the-scroll-before-it-takes-place) that provides a way to prevent a move before it takes place.

**Comment by larrybotha** [comment]
so weird, it only became an issue when I updated O.o

Thanks @alvarotrigo

**Comment by alvarotrigo** [comment]
@larrybotha its normal, the feature was introduced in fullpage.js 2.6.6 as you can see in the [change log](https://github.com/alvarotrigo/fullPage.js/releases)

**Comment by larrybotha** [comment]
haha, ok, thought it was pre-2.5.6. That clears that up. Thanks!

**Comment by el-rotny** [comment]
Great guys. Gracias.

**Comment by gwixted** [comment]
Thanks for addressing this Álvaro!

**Comment by gerhard-boden** [comment]
thanks for taking care @alvarotrigo !

**Comment by rodriguez-alex** [comment]
Hey @alvarotrigo , the fix is great! However, it now breaks silentMoveTo functionality. It always scrolls. It's in the demo provided in the documentation as well.

**Comment by alvarotrigo** [comment]
@rodriguez-alex you are right. It seems [another consequence](https://github.com/alvarotrigo/fullPage.js/issues/1442) of solving the Chrome bug... 
I didn't pay to much attention to them as I hope Chrome solves the problem for the next release. But if nothing changes for then I'll have to take a look more deeply at it.

**Comment by aktiwers** [comment]
I still seam to have this issue in Firefox 40 and Chromium Version 44.0.2403.89.

Setting css3 to false fixed it. But would love to have et true. :)

**Comment by acachary** [comment]
I have the same problem where a few sections and slides are just jumping without animating in Chrome Version 45.0.2454.101 m
It is working absolutely fine in Firefox and edge. Haven't tried it in safari yet. I tried setting 'css3: false' and that seemed to fix the problem in chrome on the desktop, but ran into trouble on mobile devices where the section just would not budge and I would end up being stuck in that section.

site: ananyaacharya.com

Desktop:
OS Name	 Microsoft Windows 10 Home
Version	10.0.10240 Build 10240

Mobile: iPhone
iOS 9.0.2

I have attached a screen recorded gif for reference.
![fullpage_bug](https://cloud.githubusercontent.com/assets/6822911/10416241/f971d1c2-6fc2-11e5-8943-b4e2bcbdb924.gif)

**Comment by alvarotrigo** [comment]
@acachary the problem with sections was solved a few versions ago. Please download the latest fullpage.js 2.7.4.
The problem with slides still taking place in the current version, please see https://github.com/alvarotrigo/fullPage.js/issues/1507

**Comment by acachary** [comment]
@alvarotrigo I have downloaded the latest release yesterday, i.e on 10/10/2015 and the gif is from the latest version.

**Comment by alvarotrigo** [comment]
@acachary are you having the problem with sections or with slides?
Can you link your site? Unless you provide a reproduction of the problem there's nothing I can do.

**Comment by acachary** [comment]
@alvarotrigo with both, a few sections and a couple of slides. http://ananyaacharya.com

**Comment by alvarotrigo** [comment]
@acachary I'm afraid I can not reproduce your issue. I don't see any problem with sections. 
Make sure you are not catching an old script for fullpage.js.

**Comment by acachary** [comment]
Thank you @alvarotrigo for the help, but I have the latest script of fullpage.js and I have tested www.ananyaacharya.com on four other machines with Chrome v45 and I'm seeing the same issue.

**Comment by alvarotrigo** [comment]
@acachary it seems you are talking about another issue such as a long scroll or a very fast scroll. Open a new issue for it please. It is definitely not the problem specified in this issue.
